03week/towersOfHanoi.js

```diff
@@ -3,52 +3,106 @@
 const assert = require('assert');
 const readline = require('readline');
 const rl = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout
+    input: process.stdin,
+    output: process.stdout
 });
 
+// This object establishes the game board
 let stacks = {
-  a: [4, 3, 2, 1],
-  b: [],
-  c: []
+    a: [4, 3, 2, 1],
+    b: [],
+    c: []
 };
 
+// This function prints the game board so it can be viewed in console
 function printStacks() {
-  console.log("a: " + stacks.a);
-  console.log("b: " + stacks.b);
-  console.log("c: " + stacks.c);
+    console.log("a: " + stacks.a);
+    console.log("b: " + stacks.b);
+    console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+// This function moves disks from one stack to another
+function movePiece(startStack, endStack) {
+    let move = stacks[startStack].pop();
+    stacks[endStack].push(move);
 }
 
-function isLegal() {
-  // Your code here
+// This function checks to see if the move is legal
+function isLegal(startStack, endStack) {
+
+    // Are the user inputs valid?  Is the peg empty?
+    let pegNames = ['a', 'b', 'c'];
```

Comment:
I like the comprehensive checks for the moves, especially the check if the stacks are empty. nice.